### PR TITLE
Fixed bug #19002: Unauthorized access to Survey menu entries 

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

File: application/controllers/admin/SurveymenuController.php [hunk 1]
```diff
@@ -367,6 +381,7 @@ protected function performAjaxValidation($model)
      **/
     public function index()
     {
+        // Permission check is done in the constructor.
```
Comment (Shnoulle) [inline application/controllers/admin/SurveymenuController.php:384]: Why in the constructoi ?

It's the only one controller where we check permission in constrictor, no ?

I think it's better to do on function OR add it too in other controller constructor (minimal access : read)

**Review by Shnoulle — COMMENTED** [review]
Why in constructor ?

File: application/controllers/admin/SurveymenuController.php (hunk 1 above)
Comment (gabrieljenik) [reply application/controllers/admin/SurveymenuController.php:384]: > It's the only one controller where we check permission in constrictor, no ?

Well, not the only place.
https://github.com/LimeSurvey/LimeSurvey/blob/master/application/controllers/admin/globalsettings.php#L32
https://github.com/LimeSurvey/LimeSurvey/blob/master/application/controllers/admin/SurveymenuEntryController.php#L13

> I think it's better to do on function OR add it too in other controller constructor (minimal access : read)

I added the comment on the functions as to try to clarify

Comment (Shnoulle) [reply application/controllers/admin/SurveymenuController.php:384]: > Well, not the only place.

OK,
then : let's go :)

**Review by Shnoulle — APPROVED** [review]

**Pull request merged**